engine/flink/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/source/flink/UniversalTimestampFieldAssigner.scala

```diff
@@ -3,15 +3,17 @@ package pl.touk.nussknacker.engine.schemedkafka.source.flink
 import org.apache.avro.generic.GenericRecord
 import org.apache.flink.api.common.eventtime.SerializableTimestampAssigner
 import org.apache.kafka.clients.consumer.ConsumerRecord
+import pl.touk.nussknacker.engine.api.typed.TypedMap
 
-object GenericRecordTimestampFieldAssigner {
+object UniversalTimestampFieldAssigner {
 
   def apply[K, V](fieldName: String): SerializableTimestampAssigner[ConsumerRecord[K, V]] = new SerializableTimestampAssigner[ConsumerRecord[K, V]] {
-    // TODO: Handle exceptions thrown within sources (now the whole process fails)
     override def extractTimestamp(element: ConsumerRecord[K, V], recordTimestamp: Long): Long =
-      Option(element.value().asInstanceOf[GenericRecord].get(fieldName))
-        .map(_.asInstanceOf[Long])
-        .getOrElse(0L) // explicit null to 0L conversion (instead of implicit unboxing)
+      element.value() match {
+        case null => 0
+        case genericRecord: GenericRecord => genericRecord.get(fieldName).asInstanceOf[Long]
+        case typedMap: TypedMap => typedMap.get(fieldName).asInstanceOf[Long]
```

Comment:
maybe also a case for `_` when schema is not of type `object` ?
e.g.
```
{
    "type": "string"
}
```

Reply:
also, maybe let's  `case map: java.util.Map`?
(probably we should just have expression instead of name, since we don't handle nested fields etc, but let's leave it for now...)